gobblin-modules/gobblin-http/src/main/java/gobblin/r2/R2RestResponseHandler.java

```diff
@@ -26,28 +29,37 @@
 public class R2RestResponseHandler implements ResponseHandler<RestRequest, RestResponse> {
 
   public static final String CONTENT_TYPE_HEADER = "Content-Type";
+  private final String R2_RESPONSE_EVENT_NAMESPACE = "gobblin.http.r2.response";
+  private final String R2_FAILED_REQUEST = "R2FailedRequest";
+  private final String REQUEST = "request";
+  private final String STATUS_CODE = "statusCode";
   private final Set<String> errorCodeWhitelist;
+  MetricContext metricsContext;
+  EventSubmitter.Builder eventSubmitterBuilder;
 
   public R2RestResponseHandler() {
-    this(new HashSet<>());
+    this(new HashSet<>(), Instrumented.getMetricContext(new State(), AsyncDataWriter.class));
   }
 
-  public R2RestResponseHandler(Set<String> errorCodeWhitelist) {
+  public R2RestResponseHandler(Set<String> errorCodeWhitelist, MetricContext metricContext) {
     this.errorCodeWhitelist = errorCodeWhitelist;
+    this.metricsContext = metricContext;
+    eventSubmitterBuilder = new EventSubmitter.Builder(metricsContext, R2_RESPONSE_EVENT_NAMESPACE);
   }
 
   @Override
   public R2ResponseStatus handleResponse(Request<RestRequest> request, RestResponse response) {
     R2ResponseStatus status = new R2ResponseStatus(StatusType.OK);
     int statusCode = response.getStatus();
     status.setStatusCode(statusCode);
-
     HttpUtils.updateStatusType(status, statusCode, errorCodeWhitelist);
 
     if (status.getType() == StatusType.OK) {
       status.setContent(response.getEntity());
       status.setContentType(response.getHeader(CONTENT_TYPE_HEADER));
     } else {
+      eventSubmitterBuilder.addMetadata(REQUEST, request.toString()).addMetadata(STATUS_CODE, String.valueOf(statusCode))
```

Comment:
I think you should bind metadata with your event instead of putting in the builder. It means:

1. eventSubmitter should only be built once. 
2. When submit, you can do eventSubmitter.submit (R2_FAILED_REQUEST, metadataMap...)